gordon/router.py

```diff
@@ -135,7 +181,13 @@ def _get_next_phase(self, event_msg):
                 #              to clean up rather than just breaking
                 break
 
+            if not await self._verify_message_impl(event_msg):
+                break
+
+            await self.metrics.incr('router-message-consumed')
+            await self._add_message_in_flight(event_msg)
             await self._route(event_msg)
+            await self._remove_message_in_flight(event_msg)
```

Comment:
Just for my clarification, why are we removing messages from being "in flight" after they are routed? Aren't messages pretty much in flight the entire time unless dropped or completed?

Reply:
Sure - we're removing from the in-memory dictionary so we can send an accurate count of messages in flight when we "set" the gauge metric.

Reply:
Maybe the confusion is from the method name (but I think `remove_message_in_flight_if_appropriate` is a bit long)? If you look in the method definition they are only removed after their final routing, just like they're only added before their first routing.